src/preprocess.ts

```diff
@@ -42,7 +42,10 @@ export default function preprocess(
 
   // We can potentially do this faster by using the language service.
   // See https://github.com/TypeStrong/ts-node/blob/master/src/index.ts#L268
-  if (tsJestConfig.enableTsDiagnostics) {
+  if (
+    tsJestConfig.enableTsDiagnostics &&
+    jestConfig.roots.find(root => filePath.indexOf(root) > -1)
```

Comment:
I believe this will work only if you choose a `rootDir` containing ONLY your test files. I think what you wanted here is something using `testMatch` (or `testRegex`, whether is set) and `testPathIgnorePatterns`. See https://jestjs.io/docs/en/configuration.html#testmatch-array-string.

Or did I misunderstood the purpose of this PR?